openwisp_controller/config/base/vpn.py

```diff
@@ -599,6 +604,15 @@ class method for ``post_delete`` signal
         except ObjectDoesNotExist:
             pass
 
+    @classmethod
+    def assign_ip(cls, instance, ip_obj, **kwargs):
+        def _assign_ip_address():
+            instance.ip = ip_obj
+            instance.full_clean()
+            instance.save()
+
+        transaction.on_commit(_assign_ip_address)
```

Comment:
as anticipated via IM, let's try to avoid the additional signal receiver but use more linear code to avoid turning this into maze runner

Reply:
Thanks to your suggesting @nemesisdesign, the code has become neater now.